Sources/SwiftFormatPrettyPrint/TokenStreamCreator.swift

```diff
@@ -2558,6 +2574,83 @@ private final class TokenStreamCreator: SyntaxVisitor {
+    // following lines in the ignored node are printed as-is with no changes to indentation.
+    var nodeText = node.description
+    if let firstToken = node.firstToken {
+      extractLeadingTrivia(firstToken)
+      let leadingTriviaText = firstToken.leadingTrivia.reduce(into: "") { $1.write(to: &$0) }
+      nodeText = String(nodeText.dropFirst(leadingTriviaText.count))
+    }
+
+    // The leading trivia of the next token, after the ignored node, may contain content that
+    // belongs with the ignored node. The trivia extraction that is performed for `lastToken` later
+    // excludes that content so it needs to be extracted and added to the token stream here.
+    if let next = node.lastToken?.nextToken, let trivia = next.leadingTrivia.first {
+      switch trivia {
+      case .lineComment, .blockComment:
+        trivia.write(to: &nodeText)
+        break
+      default:
+        // All other kinds of trivia are inserted into the token stream by `extractLeadingTrivia`
+        // when the relevant token is visited.
+        break
+      }
+    }
+
+    appendToken(.verbatim(Verbatim(text: nodeText, indentingBehavior: .firstLine)))
+  }
+
+  /// Returns whether the given trivia includes a directive to ignore formatting for the next node.
+  ///
+  /// - Parameter trivia: Leading trivia for a node that the formatter supports ignoring.
+  private func isFormatterIgnorePresent(inTrivia trivia: Trivia) -> Bool {
+     for piece in trivia {
+       switch piece {
+       case .lineComment(let text), .blockComment(let text),
+         .docLineComment(let text), .docBlockComment(let text):
+         if text.contains("swift-format-ignore") { return true }
+         break
+       default:
+         break
+       }
+     }
+     return false
+  }
+
+  /// Returns whether the formatter should ignore the given node by printing it without changing the
+  /// node's internal text representation (i.e. print all text inside of the node as it was in the
+  /// original source).
+  ///
+  /// - Note: The caller is responsible for ensuring that the given node is a type of node that can
+  /// be safely ignored.
+  ///
+  /// - Parameter node: A node that can be safely ignored.
+  private func shouldFormatterIgnore(node: Syntax) -> Bool {
+    // Regardless of the level of nesting, if the ignore directive is present on the first token
+    // contained within the node then the entire node is eligible for ignoring.
+    if let firstTrivia = node.firstToken?.leadingTrivia {
+      return isFormatterIgnorePresent(inTrivia: firstTrivia)
+    }
+
+    // Some nodes consist of multiple tokens where the user might place the ignore directive. For
+    // example, the comment could appear after a list of modifiers on a decl. Search for the ignore
+    // directive on a top level token, but don't search inside of any other children.
+    for token in node.children where token is TokenSyntax {
```

Comment:
Can you refresh my memory of a specific case where the `firstToken` is non-nil but where we might want to support the directive on other immediate children? And then can you add that example to the comment so that I don't forget it again? 🙂

Reply:
This was support situations where the ignore comment is placed between some attributes and/or modifiers on decl. For example:
```
public  // swift-format-ignore
  struct Foo {
}
```

As we discussed, I've reverted this part of the change so that the comment must be leading trivia of the first token in the decl.